src/vcpkg/configuration.cpp

```diff
@@ -679,6 +688,71 @@ namespace vcpkg
 
     Json::IDeserializer<Configuration>& get_configuration_deserializer() { return ConfigurationDeserializer::instance; }
 
+    Optional<Configuration> parse_configuration(const Filesystem& fs, const Path& path, MessageSink& messageSink)
```

Comment:
From an interface perspective, I have a mild preference to require callers to pass in the content rather than a path. Then this thing need not depend on filesystem at all.

Reply:
I like not having to repeat the file validation boilerplate, but I'll think about it.

Reply:
> I like not having to repeat the file validation boilerplate, but I'll think about it.

Almost all callers can pass `fs.read_content(path, VCPKG_LINE_INFO)` and get identical behavior.